vividus-plugin-web-app/src/main/java/org/vividus/selenium/screenshot/AshotFactory.java

```diff
@@ -37,37 +41,35 @@
 public class AshotFactory implements IAshotFactory
 {
     private ScreenshotShootingStrategy screenshotShootingStrategy;
-    private ShootingStrategy baseShootingStrategy;
 
     @Inject private IWebDriverFactory webDriverFactory;
     @Inject private IWebDriverManager webDriverManager;
     @Inject private IJavascriptActions javascriptActions;
+    @Inject private IRuntimeEnvironment runtimeEnvironment;
     @Inject private ScreenshotDebugger screenshotDebugger;
 
-    public void init()
-    {
-        baseShootingStrategy = ShootingStrategies.scaling((float) webDriverManager.getDevicePixelRatio());
-    }
+    private Supplier<ShootingStrategy> baseShootingStrategy = Suppliers.memoize(
+        () -> ShootingStrategies.scaling((float) runtimeEnvironment.getDevicePixelRatio()));
```

Comment:
is there any chance that we get bad scaling results because we convert double-precision value (as DPR stored and returned by invocation window.devicepixelratio) to single precision value?

Reply:
I've tested and haven't faced with such problem, even more for now there is no case when DPR is actually floating point number, it's always integer

Reply:
By the way, looks like it's can be floating point https://www.mydevice.io/